src/ngraph/runtime/dynamic/dynamic_backend.cpp

```diff
@@ -120,130 +120,207 @@ bool runtime::dynamic::DynamicExecutable::call(
-                wrapped_inputs.push_back(input);
+                wrapped_outputs.push_back(outputs[i]);
             }
-
-            i++;
         }
 
-        clone = specialize_function(
-            m_wrapped_function, arg_element_types, arg_shapes, arg_value_base_pointers);
+        return m_lru->get_cached_entry(merged_input_shapes)->call(wrapped_outputs, inputs);
     }
-
-    pass::Manager passes;
-    passes.register_pass<pass::ConstantFolding>();
-    passes.register_pass<pass::DynElimination>();
-    passes.register_pass<pass::Opset0Downgrade>(); // Converts dynamic v1 variants to v0 ops
-    passes.set_per_pass_validation(false);
-
-    // FIXME(amprocte): Vile, temporary hack: we need to do repeated rounds of
-    // ConstantFolding/DynElimination until everything that DynElimination is supposed to
-    // eliminate has actually been eliminated. We could do this by monitoring the return values of
-    // of the passes (keep iterating until both CF and DE report no changes), but that did not
-    // seem to work so here we are. Probably a better fix is to somehow combine the matchers in CF
-    // and DE into one pass.
-    size_t num_dyn_nodes_last_pass = std::numeric_limits<size_t>::max();
-
-    while (num_dyn_nodes_last_pass != 0)
+    else
     {
-        passes.run_passes(clone);
-        auto num_dyn_nodes_this_pass = count_dyn_nodes(clone);
+        NGRAPH_INFO << "Cache Miss";
+        NGRAPH_CHECK(m_wrapped_function->get_parameters().size() == inputs.size());
 
-        NGRAPH_CHECK(num_dyn_nodes_this_pass < num_dyn_nodes_last_pass,
-                     "Could not eliminate all Dyn nodes (",
-                     num_dyn_nodes_this_pass,
-                     " remaining)");
+        std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
+        std::vector<element::Type> arg_element_types;
+        std::vector<PartialShape> arg_shapes;
 
-        num_dyn_nodes_last_pass = num_dyn_nodes_this_pass;
-    }
+        std::shared_ptr<Function> clone;
+        {
+            // We'll use AlignedBuffers to back the base pointers, storing them in this vector for
+            // RAII
+            // purposes.
+            std::vector<AlignedBuffer> arg_buffers;
+            arg_buffers.reserve(inputs.size());
+            std::vector<void*> arg_value_base_pointers(inputs.size());
+
+            size_t i = 0;
+
+            for (auto& input : inputs)
+            {
+                if (m_wrapped_function->get_parameters()[i]->is_relevant_to_shapes())
+                {
+                    // TODO(amprocte): Move has_storage() to runtime::Tensor?
+                    if (auto dynamic_tensor =
+                            std::dynamic_pointer_cast<runtime::dynamic::DynamicTensor>(input))
+                    {
+                        NGRAPH_CHECK(dynamic_tensor->has_storage());
+                    }
+
+                    arg_buffers.emplace_back(input->get_size_in_bytes(), /*alignment=*/64);
```

Comment:
Make a veriable called alignment and set it to 64, then pass alignment into this function. It will read much better that way. Or better yet, remove the 64. Just use the default alignment.

Reply:
haven't touched this part of the code..not sure why its showing up in the diff